lib/command.js

```diff
@@ -1,62 +1,17 @@
 'use strict';
 
-var program = require ('commander');
-
+var common = require ('./common');
+var program = require ('./program');
 var doctest = require ('..');
-var pkg = require ('../package.json');
-
-
-program
-.version (pkg.version)
-.usage ('[options] path/to/js/or/coffee/module')
-.option ('-m, --module <type>',
-         'specify module system ("amd" or "commonjs")')
-.option ('    --nodejs <options>',
-         'pass options directly to the "node" binary')
-.option ('    --prefix <prefix>',
-         'specify Transcribe-style prefix (e.g. ".")')
-.option ('    --opening-delimiter <delimiter>',
-         'specify line preceding doctest block (e.g. "```javascript")')
-.option ('    --closing-delimiter <delimiter>',
-         'specify line following doctest block (e.g. "```")')
-.option ('-p, --print',
-         'output the rewritten source without running tests')
-.option ('-s, --silent',
-         'suppress output')
-.option ('-t, --type <type>',
-         'specify file type ("coffee" or "js")')
-.parse (process.argv);
 
-//  formatErrors :: Array String -> String
-function formatErrors(errors) {
-  return (errors.map (function(s) { return 'error: ' + s + '\n'; })).join ('');
-}
 
-var errors = [];
-if (program.module != null &&
-    program.module !== 'amd' &&
-    program.module !== 'commonjs') {
-  errors.push ('Invalid module `' + program.module + "'");
-}
-if (program.type != null &&
-    program.type !== 'coffee' &&
-    program.type !== 'js') {
-  errors.push ('Invalid type `' + program.type + "'");
-}
-if (errors.length > 0) {
-  process.stderr.write (formatErrors (errors));
+if (program.module === 'esm') {
+  process.stderr.write (
+    common.formatErrors (
+      ['ECMAScript modules only supported in Node version 9 and up']
```

Comment:
Could we use the following message?

_Node.js v8.9.4 does not support ECMAScript modules (supported since v9.0.0)_

_v8.9.4_ should be dynamic, of course.

Reply:
:zap: Added a test for the message as well.